src/txn/txn_timestamp.c

```diff
@@ -28,117 +27,101 @@ __wt_timestamp_to_string(wt_timestamp_t ts, char *ts_string)
+        hex_timestamp[1] = '\0';
+        return;
+    }
+    if (ts == WT_TS_MAX) {
+#define WT_TS_MAX_HEX_STRING "ffffffffffffffff"
+        (void)memcpy(hex_timestamp, WT_TS_MAX_HEX_STRING, strlen(WT_TS_MAX_HEX_STRING) + 1);
+        return;
+    }
+
+    for (p = hex_timestamp; ts != 0; ts >>= 4)
+        *p++ = (char)__wt_hex((u_char)(ts & 0x0f));
+    *p = '\0';
+
+    /* Reverse the string. */
+    for (--p; p > hex_timestamp;) {
+        v = *p;
+        *p-- = *hex_timestamp;
+        *hex_timestamp++ = v;
+    }
 }
 
 /*
  * __wt_verbose_timestamp --
  *	Output a verbose message along with the specified timestamp.
  */
 void
-__wt_verbose_timestamp(
-    WT_SESSION_IMPL *session, wt_timestamp_t ts, const char *msg)
+__wt_verbose_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t ts, const char *msg)
 {
-	char ts_string[WT_TS_INT_STRING_SIZE];
+    char ts_string[WT_TS_INT_STRING_SIZE];
 
-	__wt_verbose(session,
-	    WT_VERB_TIMESTAMP, "Timestamp %s: %s",
-	    __wt_timestamp_to_string(ts, ts_string), msg);
+    __wt_verbose(
+      session, WT_VERB_TIMESTAMP, "Timestamp %s: %s", __wt_timestamp_to_string(ts, ts_string), msg);
 }
 
 /*
  * __wt_txn_parse_timestamp_raw --
  *	Decodes and sets a timestamp. Don't do any checking.
  */
 int
-__wt_txn_parse_timestamp_raw(WT_SESSION_IMPL *session, const char *name,
-    wt_timestamp_t *timestamp, WT_CONFIG_ITEM *cval)
+__wt_txn_parse_timestamp_raw(
+  WT_SESSION_IMPL *session, const char *name, wt_timestamp_t *timestamp, WT_CONFIG_ITEM *cval)
 {
-	static const int8_t hextable[] = {
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	     0,  1,   2,   3,   4,   5,   6,   7,
-	     8,  9,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, 10,  11,  12,  13,  14,  15,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, -1,  -1,  -1,  -1,  -1,  -1,  -1,
-	    -1, 10,  11,  12,  13,  14,  15,  -1
-	};
-	wt_timestamp_t ts;
-	size_t len;
-	int hex_val;
-	const char *hex_itr;
-
-	*timestamp = 0;
-
-	if (cval->len == 0)
-		return (0);
-
-	/* Protect against unexpectedly long hex strings. */
-	if (cval->len > 2 * sizeof(wt_timestamp_t))
-		WT_RET_MSG(session, EINVAL,
-		    "%s timestamp too long '%.*s'",
-		    name, (int)cval->len, cval->str);
-
-	for (ts = 0, hex_itr = cval->str, len = cval->len; len > 0; --len) {
-		if ((size_t)*hex_itr < WT_ELEMENTS(hextable))
-			hex_val = hextable[(size_t)*hex_itr++];
-		else
-			hex_val = -1;
-		if (hex_val < 0)
-			WT_RET_MSG(session, EINVAL,
-			    "Failed to parse %s timestamp '%.*s'",
-			    name, (int)cval->len, cval->str);
-		ts = (ts << 4) | (uint64_t)hex_val;
-	}
-	*timestamp = ts;
-
-	return (0);
+    static const int8_t hextable[] = {-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
```

Comment:
😢 Should we avoid reformatting thing like this?

Reply:
Probably. I'm not 100% sure what i'm looking at but it seems important for it to be displayed as a table.